afew/tests/test_mailmover.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+# SPDX-License-Identifier: ISC
+# Copyright (c) Johannes Löthberg <johannes@kyriasis.com>
+
+import datetime
+import email
+from email.utils import format_datetime, localtime, make_msgid
+import mailbox
+import notmuch
+import os
+import shutil
+import tempfile
+import unittest
+
+
+def get_notmuch_db(db_dir):
+    return notmuch.Database(db_dir, mode=notmuch.Database.MODE.READ_WRITE)
+
+def create_mail(msg, maildir, notmuch_db, tags, old=False):
+    if old:
+        date = format_datetime(localtime() - datetime.timedelta(days=20))
+    else:
+        date = format_datetime(localtime())
+    msgid = make_msgid()
+    email_message = email.message_from_string("""\
+Date: {date}
+From: you <you@example.org
+To: me <me@example.com>
+Message-ID: {msgid}
+
+{msg}""".format(date=date, msgid=msgid, msg=msg))
+    maildir_message = mailbox.MaildirMessage(email_message)
+    message_key = maildir.add(maildir_message)
+
+    fname = os.path.join(maildir._path, maildir._lookup(message_key))
+    # notmuch 0.26 added index_file and deprecated add_message.
+    if hasattr(notmuch.Database, 'index_file'):
+        (notmuch_msg, status) = notmuch_db.index_file(fname)
+    else:
+        (notmuch_msg, status) = notmuch_db.add_message(fname)
+    for tag in tags:
+        notmuch_msg.add_tag(tag, False)
+
+    return (msgid[1:-1], msg)
+
+
+class TestMailMover(unittest.TestCase):
+    def setUp(self):
+        self.test_dir = tempfile.mkdtemp()
+        os.environ['MAILDIR'] = self.test_dir
+
+        self.notmuch_config = os.path.join(self.test_dir, 'notmuch-config')
+        with open(self.notmuch_config, 'w+') as fp:
```

Comment:
There's `afew/NotmuchSettings.py` - Maybe we can add this code there?

Reply:
Hm, you mean like some way to pass it a dict and then have that written as the config?  Could do that, though I'm not sure if it would be used by anything other than this one place in the code.

Reply:
Added `write_notmuch_settings`, and just set `notmuch_settings` instead of doing it in a more roundabout way.